client/src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import FocusLock from "react-focus-lock";
+
+import "./Sidebar.scss";
+
+import { CSSTransition } from "react-transition-group";
+
+import { trivial_text_maker } from "src/models/text";
+
+import { IconX } from "src/icons/icons";
+
+interface SidebarProps {
+  is_open: boolean;
+  callback: (value: boolean) => void;
```

Comment:
More descriptive name would be handy, like `open_close_callback` (of even splitting in to distinct open and close callbacks).